pkg/storage/bolt.go

```diff
@@ -14,9 +14,14 @@ import (
 )
 
 const (
-	DBFile = "ssi-service.db"
+	Bolt Storage = "bolt"
 )
 
+// try to register your implementation at your .go file like this:
+func init() {
+	RegisterStorage(new(BoltDB))
+}
```

Comment:
the problem I have with this is - if I am running this service in production, and want to use another storage provider...do I now have multiple storages running at the same time?

i could see a world where there are multiple storage providers...but as written now the code assumes one. does introduce a path where that constraint is broken?

Reply:
`RegisterStorage(new(BoltDB))` just make a pointer of  bolt storage. like write a selection on paper. the real process who make and initialize the bolt storage is `storage.Init(interface{})` at `NewStorage()`. 
so after `RegisterStorage ` done, we still have no useable storage instance. At this time we only have a **canability* *that we can use `NewStorage` to choice a storage and make it useable.
For example, `ssi-sevice` have a `default` storage selection, it's `Bolt`. Developer add  `LevelDB` and `Badger` to service with `RegisterStorage`. now he can select use `Bolt`, `LevelDB` or `Badger` with calling `NewStorage`, if he selected `Badger`, `NewStorage` get `Badger` from the`valiableStorage`, create or open db file in `storage.Init(interface{})`. now `Badger` is ready to use.

Reply:
thanks this makes sense